File: FEB_2021/Year.java
import java.io.*;
import java.util.*;
public class Year {
    public static int N;
    public static int K;
    public static ArrayList<Integer> t;
    public static int[] p;
    public static int[] diff;
    public static int time=0;
    public static void main(String[] args) throws IOException{
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        String[] store = r.readLine().split(" ");
        N = Integer.parseInt(store[0]);
        K = Integer.parseInt(store[1]);
        t = new ArrayList<Integer>();
        for (int i=0;i<N;i++){
            int stores = Integer.parseInt(r.readLine());
            boolean done = false;
            for (int j=0;j<t.size();j++){
                if (t.get(j)==(stores-(stores%12))){
                    done = true;
                }
            }
            if (!done){
                t.add(stores-(stores%12));
            }
        }
        Collections.sort(t);
        time+=t.size()*12;
        diff = new int[t.size()];
        for (int i=0;i<t.size()-1;i++){
            diff[i] =t.get(i+1)-t.get(i);
        }
        Arrays.sort(diff);
        for (int i=0;i<t.size()-K+1;i++){
            time+=diff[i];
        }
        System.out.println(time);
    }
}

File: FEB_2021/c++/Year.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;
vector<int> y;
int main(){
    int N, K;
    cin >> N >> K;
    for (int i=0;i<N;i++){
        int store;
        cin >> store;
        y.push_back((store/12)*12);
    }
    sort(y.begin(),y.end());
    int d[N];
    for (int i=0;i<N-1;i++){
        d[i]=(y[i+1]-y[i]); 
    }
    cout << endl;
    for (int i=0;i<N-1;i++){
        cout << d[i] << " ";
    }
    cout << endl;
    sort(d,d+N-1);
    int min = pow(10,9);
    for (int j=2;j<=K;j++){
        int year = 0;
        year += 12*(N-j+1);
        for (int i=0;i<N-j+1;i++){
            year+=d[i];
        }
        if (year<min){
            min=year;
        }
    }
    cout << min;
}